apps/token/app/src/token.rs

```diff
@@ -135,25 +209,63 @@ impl Token {
         let new_from_balance = self.calculate_from_balance(key, amount).await?;
 
         let state = self
-            .query(&self.deployed_predicates.token, &query_balances(to.into()))
+            .query(
+                &self.deployed_predicates.token,
+                &token::query_balances(to.into()),
+            )
             .await?;
         let to_balance = state.first().copied().unwrap_or_default();
         let Some(new_to_balance) = to_balance.checked_add(amount) else {
             bail!("Overflow error")
         };
 
-        let builder = SignedTransfer {
-            auth_address: self.deployed_predicates.signed_transfer.clone(),
-            token_address: self.deployed_predicates.transfer.clone(),
-            from_account_name: from_name.to_string(),
-            to_account_name: to_name.to_string(),
-            new_nonce,
-            amount,
-            new_from_balance,
-            new_to_balance,
+        const AUTH_PATH: Word = 0;
+        const TRANSFER_PATH: Word = 1;
+
+        let mut data = key.to_vec();
+        data.extend(to);
+        data.push(amount);
+        data.push(new_nonce);
+        data.extend(self.deployed_predicates.transfer.contract.encode());
+        data.extend(self.deployed_predicates.transfer.predicate.encode());
+
+        let mut solution = Solution {
+            data: Default::default(),
+        };
+
+        let auth = signed::TransferData {
+            predicate_to_solve: self.deployed_predicates.signed_transfer.clone(),
+            decision_variables: signed::Transfer::Vars {
+                ___I_pathway: TRANSFER_PATH,
+                sig: self.sign_data(from_name, data)?.encode(),
+            },
+            transient_data: signed::Transfer::pub_vars::mutations().token(|addr| {
+                let (c, p) = self.deployed_predicates.transfer.encode();
+                addr.contract(c).addr(p)
+            }),
         };
 
-        let solution = builder.build(&mut self.wallet)?;
+        solution.data.insert(AUTH_PATH as usize, auth.into());
+
+        let transfer = token::TransferData {
+            predicate_to_solve: self.deployed_predicates.transfer.clone(),
+            decision_variables: token::Transfer::Vars {
+                auth_addr: self.deployed_predicates.signed_transfer.encode(),
+                ___A_pathway: AUTH_PATH,
+            },
+            transient_data: token::Transfer::pub_vars::mutations()
+                .key(key)
+                .to(to)
+                .amount(amount),
+            state_mutations: token::storage::mutations()
+                .balances(|map| map.entry(key, new_from_balance))
+                .balances(|map| map.entry(to, new_to_balance))
+                .nonce(|nonces| nonces.entry(key, new_nonce)),
+        };
+
+        solution
+            .data
+            .insert(TRANSFER_PATH as usize, transfer.into());
 
         // Submit the solution
```

Comment:
```suggestion
```
Same for line 124 (currently)

Reply:
Why is this comment bad? Isn't that what's going on below?